_resources/port1.0/group/fuse-1.0.tcl

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8; mode: tcl; c-basic-offset: 4; indent-tabs-mode: nil; tab-width: 4; truncate-lines: t -*- vim:fenc=utf-8:et:sw=4:ts=4:sts=4
+#
+# This PortGroup chooses the right FUSE library to use.
+#
+# Usage:
+#
+# PortGroup           fuse 1.0
+#
+
+# use osxfuse up to OS X 10.11
+if {${os.platform} eq "darwin" && ${os.major} < 16} {
```

Comment:
Since we now support a cask-style binary, my recommendation is that we select the optimal one for each macOS release. This is per the discussion we had for [PR 10620](https://github.com/macports/macports-ports/pull/10620):

> IMHO, if we're going to utilize the binary for this port, then we should select the "best" version for a given MacOS release. And it's only three cases - 3.11.2 for 10.5 through 10.8, 4.0.5 for 10.9 through 10.11, and the latest (currently 4.1.0) for 10.12+.

Since all binary releases of macFUSE are newer than the old one we still have [prior to the project switching to a closed-source model] - and there are likely numerous bug fixes, which would benefit all macOS releases, even those as old as 10.5 - my suggestion is that we auto-select the binary macFUSE version.

Then add support for a variant, allowing users to select the legacy open-source version, on macOS releases that support it.

Reply:
I don't catch what do you mean with "auto-select the optimal". My idea was to use opensource whenever possible and then use binary/blob. Anyway, the choose between 4.0.5 or 4.1.0 must be done at macfuse port level and not in the portgroup. You are right about fixes/improvement but I personally prefer, as I said, to use the opensource one before everything else.

Variant: no. If you want to install the binary or the opensource you can still install it manually before installing the fuse module.

Reply:
can you answer?

Reply:
> I don't catch what do you mean with "auto-select the optimal". My idea was to use opensource whenever possible and then use binary/blob. Anyway, the choose between 4.0.5 or 4.1.0 must be done at macfuse port level and not in the portgroup. You are right about fixes/improvement but I personally prefer, as I said, to use the opensource one before everything else.

Ah yes, you're right: the auto-selection [for binary version] should occur in port `macfuse`.

> Variant: no. If you want to install the binary or the opensource you can still install it manually before installing the fuse module.

Fair enough.